boards: cy8cproto_041tp: i2c support

manoj-aerlync opened the pull request:
I2C Driver for cy8cproto_041tp.This PR has dependency on https://github.com/zephyrproject-rtos/zephyr/pull/100894

manoj-aerlync committed:
drivers: I2C: Add Infineon PSOC4 I2C slave support

De-initialize the SCB when switching between I2C master
and slave modes

Signed-off-by: Manojkumar Konisetty <manoj@aerlync.com>
Signed-off-by: Sayooj K Karun <sayooj@aerlync.com>

manoj-aerlync committed:
samples: sensor: bme280: boards: Add overlay file

- Add an overlay to enable bme280 sensor sample.
- Contains configurations for I2C bus and Clock

Signed-off-by: Manojkumar Konisetty <manoj@aerlync.com>
Signed-off-by: Sayooj K Karun <sayooj@aerlync.com>

manoj-aerlync committed:
drivers: i2c: update Infineon PDL I2C driver

Implement I2C configure() handling in the driver
to apply speed settings and default hardware options.

Signed-off-by: Manojkumar Konisetty <manoj@aerlync.com>
Signed-off-by: Sayooj K Karun <sayooj@aerlync.com>

manoj-aerlync committed:
dts: infineon: add I2C pins for PSoC4100TP

Include package-level I/O mapping for I2C pins
in psoc4100tp.64-tqfp.dtsi

Signed-off-by: Manojkumar Konisetty <manoj@aerlync.com>
Signed-off-by: Sayooj K Karun <sayooj@aerlync.com>

manoj-aerlync committed:
boards: cy8cproto_041tp: enable I2C

- Add pin control definitions with proper open-drain drive mode
  for SCL/SDA in cy8cproto_041tp-pinctrl.dtsi.
- Add I2C node and clock configuration to cy8cproto_041tp.dts.

Signed-off-by: Manojkumar Konisetty <manoj@aerlync.com>
Signed-off-by: Sayooj K Karun <sayooj@aerlync.com>

sreeramIfx reviewed (commented):
Looks good to me mostly. Compliance checks need to be fixed

JarmouniA commented:
> This PR has dependency on #100894

Then it should stay as a draft until the dependency is merged and it's rebased on it.

teburd commented on drivers/i2c/i2c_infineon_pdl.c:
```diff
@@ -318,6 +325,7 @@ static int ifx_cat1_i2c_configure(const struct device *dev, uint32_t dev_config)
 	const struct ifx_cat1_i2c_config *config = dev->config;
 	cy_en_scb_i2c_status_t rslt;
 	int ret;
+	bool is_slave_mode = false;
```
```suggestion
	bool is_target_mode = false;
```

manoj-aerlync committed:
samples: sensor: mpu6050: boards: Add overlay file

- Add an overlay to enable mpu6050 sensor sample.
- Contains configurations for I2C bus and Clock

Signed-off-by: Manojkumar Konisetty <manoj@aerlync.com>
Signed-off-by: Sayooj K Karun <sayooj@aerlync.com>

teburd reviewed (changes requested):
Some minor changes I'd like to see, should be quick to do.

teburd commented on drivers/i2c/i2c_infineon_pdl.c:
```diff
@@ -517,7 +541,7 @@ static int ifx_cat1_i2c_transfer(const struct device *dev, struct i2c_msg *msg,
 
 		/* Acquire semaphore (block I2C async transfer for another thread) */
-		ret = k_sem_take(&data->transfer_sem, K_FOREVER);
+		ret = k_sem_take(&data->transfer_sem, I2C_TRANSFER_TIMEOUT_MSEC);
```
I’m aware of what happens. I do not see anything here to recover. If there’s no recovery it’s better to fail hard and fail fast.
